tools/shared.py

```diff
@@ -2947,7 +2947,17 @@ def check_binaryen(bindir):
       output = run_process([finalize, '--version'], stdout=PIPE).stdout
     except subprocess.CalledProcessError:
       exit_with_error('error running binaryen executable (%s). Please check your binaryen installation' % finalize)
-    version = output.split()[2]
+    try:
+      version = output.split()[2]
+    except IndexError:
+      warning('could not check binaryen version: %s (expected %s)', output, EXPECTED_BINARYEN_VERSION)
+      return
+    # handle versions like "(version_90-45-gffe7123"
+    if version[0] == '(':
+      version = version[1:]
+    version = version.replace('version_', '')
+    if '-' in version:
+      version = version.split('-')[0]
```

Comment:
No need to handle version like this since we didn't start checking until version 90 which has the correct format.

Reply:
Hmm, I needed every single line here in this PR to fix things locally for me.

Is this the correct format?
```bash
$ bin/wasm-opt --version
wasm-opt version  (version_90-45-gffe7123)
```

Reply:
Oh I see.  It should say "wasm-opt version 90 (version_90-45-gffe7123)".

Can you see how you ended up with an empty string for the 3rd element?

Reply:
The only way I can see that happening if `CMAKE_PROJECT_VERSION` was set the empty string.

Can you add `message("${CMAKE_PROJECT_VERSION}")` to your CMakeLists.txt file to see if this is happening?

Reply:
I'll have to do it when I get home, as it it only happens there. My guess is it's using an ancient CMake version there... anyhow, this PR was necessary to fix things there, so I don't think we'll learn anything that would change the necessity of the PR.

Reply:
I don't want to have the complexity of trying to parse whats inside the braces.  All valid, recent, versions of binaryen should report they version in colum 3:

```
wasm-opt version <XXXX> (YYY)
```

We shouldn't try to be clever and handle the case where XXX is missing.  This is a bug in the building of binaryen that we need to fix.

Reply:
I'll take a look at old versions of cmake.  I could be that we need to bump the binaryen build requs a little.

Reply:
binaryen currently requires cmake `3.1.3` and the version attribute on `project` was apparently introduced in `3.0` so I can't see how that could be the issue ... strange.

Reply:
I confirmed on my home machine that `${CMAKE_PROJECT_VERSION}"` is the empty string. (I don't know enough cmake to guess why that is.)

Reply:
What version of cmake are you using? 

Try a clean build?

Reply:
Once again, I'll try when I get home :)